### ISSUE TITLE
nerdctl compose fails with warning: "Ignoring: volume: Bind: [CreateHostPath]"

### ISSUE THREAD

**Issue by mirzaprangon** [rancher-sandbox/rancher-desktop#3603]
### Discussed in https://github.com/rancher-sandbox/rancher-desktop/discussions/3150

<div type='discussions-op-text'>

<sup>Originally posted by **mirzaprangon** October 13, 2022</sup>
Hello
I am on windows, rancher desktop version 1.6.0.
I have a simple compose file with directory mounts, nothing fancy just a mysql container.

```yaml
services:
  mysql_db:
    image: mysql:8.0.30
    restart: always
    volumes:
      - ./mysql_db_data:/var/lib/mysql
      - ./mysql_db_config:/etc/mysql/conf.d
    environment:
      - MYSQL_DATABASE=keycloak
      - MYSQL_ROOT_PASSWORD=root_password
      - MYSQL_USER=db_user
      - MYSQL_PASSWORD=db_password
      - TZ=Asia/Tokyo
    command: 
      - --sql_mode=ANSI,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION
      - --lower-case-table-names=1
      - --default-time-zone=UTC
    ports:
      - 3306:3306
```

**But `nerdctl compose up` fails if I run with containerd engine**. The output I get is as follow
```log
WARN[0000] Ignoring: volume: Bind: [CreateHostPath]
WARN[0000] Ignoring: volume: Bind: [CreateHostPath]
INFO[0000] Ensuring image mysql:8.0.30
INFO[0000] Ensuring image quay.io/keycloak/keycloak:19.0.3
INFO[0000] Creating container development_environment_mysql_db_1
INFO[0000] Creating container development_environment_keycloak_1
FATA[0000] error while creating container development_environment_keycloak_1: exit status 1
```

I can run the compose file if I change my container engine to `Dockerd/moby` and use `docker compose up`.
If I remove the volume binds I can run the compose file with `nerdctl compose up`.

I am on windows. Outut of `rdctl list-settings`
```json
{
  "version": 4,
  "kubernetes": {
    "version": "1.24.4",
    "memoryInGB": 2,
    "numberCPUs": 2,
    "port": 6443,
    "containerEngine": "containerd",
    "checkForExistingKimBuilder": false,
    "enabled": false,
    "WSLIntegrations": {},
    "options": {
      "traefik": true,
      "flannel": true
    },
    "suppressSudo": false,
    "hostResolver": true,
    "experimental": {
      "socketVMNet": false
    }
  },
  "portForwarding": {
    "includeKubernetesServices": false
  },
  "images": {
    "showAll": true,
    "namespace": "buildkit"
  },
  "telemetry": true,
  "updater": true,
  "debug": false,
  "pathManagementStrategy": "notset",
  "diagnostics": {
    "showMuted": false,
    "mutedChecks": {}
  }
}
```

`nerdctl compose` Should make empty mount directories if those do not exist like `docker compose`</div>

**Comment by AkihiroSuda** [comment]
https://github.com/containerd/nerdctl/issues is the best place to report this

**Comment by mirzaprangon** [comment]
Created a issue in nerdctl repository.

**Comment by iainnicol** [comment]
I presume this issue can be closed, or at least when Rancher Desktop updates its dependencies. Akihiro Suda recently fixed this in nerdctl (thanks!).

**Comment by mirzaprangon** [comment]
As far as I can see the fix is scheduled for nerdctl `v 1.3.1`. Rancher desktop(I am on windows) 1.8.1 ships nerdctl 1.2.1 with it.

**Comment by AkihiroSuda** [comment]
> I presume the fix is scheduled for nerdctl v 1.3.1. Milestone 1.3.1 is attached to the PR.

Yes. https://github.com/containerd/nerdctl/releases/tag/v1.3.1

**Comment by mook-as** [comment]
Closing as we've had nerdctl 1.3.1 since Rancher Desktop 1.9.0 (June 2023).
